### DISCUSSION TITLE
Is there a documentation on how to run a simple example with ```wasmi```?

### DISCUSSION THREAD

**Discussion by djolertrk** [wasmi-labs/wasmi#595] [Q&A] [answered]
Hello,

I am wondering, is there a documentation on how to interpret a simple ```.wasm``` file, for example the https://github.com/paritytech/ink#hello-world--the-flipper with this tool? :)
IIUIC, the idea of the interpreter, among the other goals, is to be able to interpret ```.wasm``` files "offline", right?

Thanks in advance!

**Comment by Robbepop** [comment, marked as answer]
The `wasmi` interpreter repository provides a `wasmi_cli` application out of the box that allows to run raw Wasm files and soon WASI compatible Wasm files with the following command when your work directory is in the `wasmi` repository:
```
cargo run --package wasmi_cli <WASM_FILE> <FUNCTION_NAME> [<FUNC_ARGS>]*
```
Where `WASM_FILE` is the Wasm file path, `FUNCTION_NAME` is the name of the exported function of the Wasm file you want to call and `[<FUNC_ARGS>]*` is the optional number of arguments for the called function.

However, in your special case this tool won't help you since you apparently want to run ink! smart contracts using `wasmi`. However, ink! smart contracts require a special environment that is defined by the [`contracts-pallet`](https://github.com/paritytech/substrate/tree/master/frame/contracts). WebAssembly only defines an abstract virtual machine but what you want is a concrete invokation of a specific one - namely the Substrate contracts Wasm VM.

Therefore your best bet is to use the [ink! playground](https://ink-playground.substrate.io/) to toy around a bit or alternative use a proper [Substrate smart contracts UI](https://contracts-ui.substrate.io/) to make use of a deployed chain.

**Reply by djolertrk** [reply]
Ahh, OK. Thanks a lot anyway! I thought I could do it without a chain.

Basically, I am interested in seeing what would be the scenario with the program if it crashes (for example read from a 0x0 mem address) - more precisely, I am wondering if there is a "coredump" like file that is being generated by the WASM VM.

**Reply by Robbepop** [reply]
Actually, there are some ink! smart contract instantiation benchmarks for `wasmi` that might give you an idea of what is needed to make them run. However, I guess that they are not that useful without context: https://github.com/paritytech/wasmi/blob/master/crates/wasmi/benches/benches.rs#L134
